hmcts/idam-web-public
# feat(SIDM-3294-otp-chain): otp authentication flow

### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/SIDM-3294


### Change description ###
NOT IN SCOPE: validation

- update verifyPolicies method to support "mfa required" flag
- update /login endpoint to support mfa flow:
  - when "mfa required" then "initiate auth flow":
    - set `Idam.AuthId` cookie
    - redirect user to otp verifiction page
- update POST /verification to support submitting OTP code
  - submit OTP code along with Idam.AuthId cookie
  - grab Idam.Session cookie
  - call authorize using the Idam.Session cookie from the otp flow
  - redirect user according to the authorize call


**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```
